From 2dfe4236b57c85a87b45a1fbfa82ab031078dc76 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Thu, 17 Jun 2021 17:30:33 +0800
Subject: [PATCH] [riscv64] Fix mjsunit/regexp-global fatal error

Change-Id: Ic36d34ca928b2dbc7427f60818dbd612b386e7a2
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/2967709
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Commit-Queue: Ji Qiu <qiuji@iscas.ac.cn>
Cr-Commit-Position: refs/heads/master@{#75211}
---
 src/regexp/riscv64/regexp-macro-assembler-riscv64.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/regexp/riscv64/regexp-macro-assembler-riscv64.cc b/src/regexp/riscv64/regexp-macro-assembler-riscv64.cc
index cc48c840d7..932e045848 100644
--- a/src/regexp/riscv64/regexp-macro-assembler-riscv64.cc
+++ b/src/regexp/riscv64/regexp-macro-assembler-riscv64.cc
@@ -797,7 +797,7 @@ Handle<HeapObject> RegExpMacroAssemblerRISCV::GetCode(Handle<String> source) {
         // output registers is reduced by the number of stored captures.
         __ Sub64(a1, a1, num_saved_registers_);
         // Check whether we have enough room for another set of capture results.
-        __ BranchShort(&return_a0, lt, a1, Operand(num_saved_registers_));
+        __ Branch(&return_a0, lt, a1, Operand(num_saved_registers_));
 
         __ Sd(a1, MemOperand(frame_pointer(), kNumOutputRegisters));
         // Advance the location for output.
-- 
2.35.1

